azerothcore/azerothcore-wotlk
# fix(scripts/Gruul): Boss script fine tuning

## Changes Proposed:
- Hurtful Strike can now hit the highest aggro target, if no other targets are in range.
- Gruul is now stuck in place between Ground Slam and Shatter and does not hit.
- Gronn Lord's Grasps stack up a bit earlier


## Issues Addressed:
- Gruul does not attack anyone with Hurtful Strike, if no other targets than the MT were in range.
- Gruul chases his target between Ground Slam and Shatter.
- Gronn Lord's Grasps stacked up too late, resulting in the player being stuck too long after Shatter


## Tests Performed:
Built and tested with Win10 64bit


## How to Test the Changes:
.cheat god
.tele gruulslair
Enter the instance and reach Gruul
- Check combat log, how he does apply Hurtful Strike to the MT if no other melee targets are in range.
- Watch how Gruul stays in place between Ground Slam and Shatter
- Look at the debuffs and see how the Grasp wears off at the same time like it did in retail compared to the cast of "Shatter" and also "Ground Slam"


## Known Issues and TODO List:
There seems to be a bit of a timing problem regarding the Gronn Lords Grasp wearing off later than in original videos. 
( commit 1236344d05abcafa644ac86f60cc5aac0025563d which is part of this PR should solve that, too )

For comparison i used:
[Video of this PRs initial commit](https://streamable.com/yf98ys) 
[Youtube 1](https://youtu.be/YzSiTtbWREI?t=183) (Not 1.0 speed)
[Youtube 2](https://youtu.be/Rg4x9CRk4TA?t=54)
[Video of this PR with 4 commits](https://streamable.com/x7we1n)

## Target Branch(es):


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
